app/models/shelter.rb

```diff
@@ -4,6 +4,8 @@ class Shelter < ApplicationRecord
   validates :city, presence: true
 
   has_many :pets, dependent: :destroy
+  has_many :pet_applications, through: :pets
+  has_many :applications, through: :pet_applications
```

Comment:
Excellent job leveraging the associations that pet has set up in the shelter model!! This makes the join to applications muuuuch easier